app/src/main/java/com/example/next_show/data/RecommendationClient.java

```diff
@@ -38,59 +40,48 @@ public RecommendationClient(Shows showsObj) {
         this.showsObj = showsObj;
     }
 
-    public void fetchRecommendedShows(List<String> savedShows, FeedFragment.ShowCallback showCallback, FeedFragment.GenreMatchedCallback gmCallback) {
-        this.genreMatchedCallback = gmCallback;
-        this.showCallback = showCallback;
-
-        if (savedShows.isEmpty()){
-            // if no shows to compare, call user genre matched ones
-            fetchGenreMatchedShows(showsObj, genreMatchedCallback);
+    public void fetchRelatedShows(List<String> savedShows, ResponseCallback callback) {
+        // check if no saved shows and immediately fail
+        if(savedShows.isEmpty() || savedShows == null) {
+            callback.onFailure(NO_CALL);
             return;
         }
 
-        // get related shows based on saved LIKED shows -> updates adapter within callback
-        fetchRelatedShows(showsObj, savedShows, showCallback);
-    }
-
-    private void fetchRelatedShows(Shows traktShows, List<String> savedShows, ResponseCallback callback) {
         // use randomly picked show ID to generate related, recommended shows
         final int randomShowIndex = new Random().nextInt(savedShows.size());
         String searchID = savedShows.get(randomShowIndex);
         Log.i(TAG, "Chosen: " + searchID);
 
         try {
             // enqueue to do asynchronous call and execute to do it synchronously
-            traktShows.related(searchID, PAGES_REQUESTED, LIMIT, Extended.FULL).enqueue(new Callback<List<com.uwetrottmann.trakt5.entities.Show>>() {
+            showsObj.related(searchID, PAGES_REQUESTED, LOWER_LIMIT, Extended.FULL).enqueue(new Callback<List<com.uwetrottmann.trakt5.entities.Show>>() {
                 @Override
                 public void onResponse(Call<List<com.uwetrottmann.trakt5.entities.Show>> call, Response<List<com.uwetrottmann.trakt5.entities.Show>> response) {
                     if (response.isSuccessful()) {
                         callback.onSuccess(Show.formatShows(response.body()));
                     } else {
                         callback.onFailure(response.code());
-
-                        // only if this first call fails...
-                        // get recommended shows based on User preferences -> updates adapter within callback
-                        fetchGenreMatchedShows(showsObj, genreMatchedCallback);
-                        return;
                     }
                 }
 
                 @Override
                 public void onFailure(Call<List<com.uwetrottmann.trakt5.entities.Show>> call, Throwable t) {
                     Log.e(TAG, "OnFailure", t);
+                    callback.onFailure(NO_CALL);
```

Comment:
So, you did make the call right?  There was just some throwable here.  I think you would want to use a different error code here